src/Storages/ObjectStorage/registerStorageObjectStorage.cpp

```diff
@@ -262,6 +262,34 @@ void registerStorageDeltaLake(StorageFactory & factory)
             .source_access_type = AccessType::S3,
             .has_builtin_setting_fn = StorageObjectStorageSettings::hasBuiltin,
         });
+    factory.registerStorage(
+        "DeltaLakeS3",
+        [&](const StorageFactory::Arguments & args)
+        {
+            auto configuration = std::make_shared<StorageS3DeltaLakeConfiguration>();
+            return createStorageObjectStorage(args, configuration, args.getLocalContext());
+        },
+        {
+            .supports_settings = false,
```

Comment:
I think we should set `support_settings` for all data lake types to `true`, because there is `StorageObjectStorageSettings.cpp` which should support the settings for data lakes as well. And Judging by the fact that `StorageObjectStorageSettings` have only one setting `allow_dynamic_metadata_for_data_lakes` - it is expected that data lake storages would be able to support it. I see there is a test for this use case for iceberg https://github.com/ClickHouse/ClickHouse/blob/fb1b312611624b1d5d23077437bacd03a939299b/tests/integration/test_storage_iceberg/test.py#L207
It should work for `deltaLake` as well.

Reply:
addressed it